org-roam.el

```diff
@@ -105,39 +125,81 @@ Valid states are 'visible, 'exists and 'none."
+                                (puthash link updated backlinks)
+                                (puthash link updated-contents content)))
                             (with-current-buffer (current-buffer)
-                              (org-element-map (org-element-parse-buffer) 'link
-                                (lambda (link)
-                                  (let ((type (org-element-property :type link))
-                                        (path (org-element-property :path link)))
-                                    (when (and (string= type "file")
-                                               (string= (file-name-extension path) "org"))
-                                      path))))))))
-                org-roam-files)
-        (prin1-to-string backlinks)))
-   (lambda (backlinks)
-     (setq org-roam-hash-backlinks (car (read-from-string
-                                         backlinks)))
+
+                              (let* ( (lst (org-element-map (org-element-parse-buffer) 'link
+                                             (lambda (link)
+                                               (let* ((type (org-element-property :type link))
+                                                      (path (org-element-property :path link))
+                                                      (end (org-element-property :end link))
+                                                      )
+                                                 (when (and (string= type "file")
+                                                            (string= (file-name-extension path) "org"))
+                                                   (goto-char end)
+
+
+                                                   (let*
+                                                       ((heading (progn
+                                                                   (unless (org-before-first-heading-p)
+                                                                     (outline-back-to-heading))
+                                                                   (org-element-at-point) ))
+                                                        (end (if
+                                                                 (org-element-property :contents-end heading)
+                                                                 (org-element-property :contents-end heading)
+                                                               (org-element-property :end heading))))
+                                                     (list (file-name-nondirectory path)
+                                                           (buffer-substring
+                                                            (org-element-property :begin heading)
+                                                            end))))))
+                                             nil nil nil nil))
+                                      (res lst))
+                                (print res)
```

Comment:
I suppose this was a debug statement?

Reply:
> Nice, thanks! Haven't tried it out, just leaving some comments here first.
> 
> This is definitely on my roadmap. I'm thinking of using [Magit-Section](https://magit.vc/manual/magit-section/) to implement the org-roam side-buffer, but your work is a great starting point.
> 

Awesome! I didn't know magit-section existed. I remember 'borrowing' a lot of that code years back when I did https://github.com/ryjm/habitrpg.el - good to know they've made the library public. 

> I have a paper deadline, so I'm tied down for the next 2 weeks, but I'll give this a proper look when that's done!

No rush!